prov/shm/src/smr_msg.c

```diff
@@ -38,18 +38,18 @@
 #include "smr.h"
 
 
-static ssize_t smr_generic_recvmsg(struct fid_ep *ep_fid, const struct iovec *iov,
+static ssize_t smr_generic_recvmsg(struct smr_ep *ep, const struct iovec *iov,
 				   size_t iov_count, fi_addr_t addr, uint64_t tag,
 				   uint64_t ignore, void *context, uint64_t flags)
 {
-	struct smr_ep *ep;
 	struct smr_queue *recv_queue;
 	struct smr_ep_entry *entry;
 	ssize_t ret;
 
 	assert(iov_count <= SMR_IOV_LIMIT);
+	if (flags & FI_MULTI_RECV)
+		assert(iov_count == 1);
```

Comment:
This can be converted to:
```
assert(!(flags & FI_MULTI_RECV) || (iov_count == 1))
```
to avoid an empty if statement on non-debug builds.